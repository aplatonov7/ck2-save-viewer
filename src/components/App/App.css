.App {
  height: 100vh;
  background-color: #eee;

  display: flex;
  flex-direction: column;
}

.App__container {
  position: relative;

  display: flex;
  align-items: center;
  justify-content: center;

  flex: 1;
}

.App__data-loaded .App__container {
  padding-left: 250px;
}