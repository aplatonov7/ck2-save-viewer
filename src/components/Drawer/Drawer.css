.Drawer__delete-btn {
  border: none;
  background: none;
  cursor: pointer;
  padding: .2rem .5rem;
  background-color: #eee;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  font-size: 0;
}

.Drawer__delete-btn:after {
  content: 'x';
  font-size: 1rem;
}

.Drawer__row {
  display: flex;
  justify-content: space-between;
}

.Drawer {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  width: 500px;
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);

  padding: 1rem;

  transform: translateX(-502px);
  transition: transform .2s ease; 
}

.Drawer.Drawer__opened {
  transform: translateX(0);
}