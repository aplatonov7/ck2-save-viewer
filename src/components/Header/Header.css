.Header {
  padding: 1rem;
  background-color: #3f51b5;
  color: #fff;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  z-index: 5;
}