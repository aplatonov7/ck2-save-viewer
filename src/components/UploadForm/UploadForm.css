.UploadForm__container {
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  background-color: #fff;
}

.UploadForm__container-loading {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  padding: 2rem;
}

.UploadForm__label {
  display: block;
  padding: 2rem;
  cursor: pointer;
}

.UploadForm__input {
  display: none;
}

.UploadForm__caption {
  margin: 1rem 1rem .5rem;
}

.UploadForm__sub-caption {
  margin: 0 1rem 1rem;
}