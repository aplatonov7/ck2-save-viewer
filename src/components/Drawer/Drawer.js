import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import './Drawer.css'

const Drawer = ({ opened, onClear }) => (
  <aside className={cn('Drawer', { Drawer__opened: opened })}>
    <div className="Drawer__row">
      Data is loaded
      <button
        className="Drawer__delete-btn"
        title="Delete current data so you can upload another file"
        onClick={onClear}
      >
        Delete data
      </button>
    </div>
  </aside>
)

Drawer.propTypes = {
  opened: PropTypes.bool.isRequired,
  onClear: PropTypes.func.isRequired,
}

export default Drawer
