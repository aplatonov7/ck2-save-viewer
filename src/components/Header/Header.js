import React from 'react'

import './Header.css'

const Header = () => (
  <header className="Header">
    CK2 save to stats
  </header>
)

export default Header
