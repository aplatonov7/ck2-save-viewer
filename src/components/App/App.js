import React, { Component } from 'react'
import cn from 'classnames'

import Drawer from '../Drawer'
import Header from '../Header'
import UploadForm from '../UploadForm'

import './App.css'

class App extends Component {
  constructor(...props) {
    super(...props)

    this.state = {
      isDataLoaded: false,
    }
  }

  onDataLoad = () => {
    this.setState(() => ({ isDataLoaded: true }))
  }

  onDataClear = () => {
    this.setState(() => ({ isDataLoaded: false }))
  }

  render() {
    const { isDataLoaded } = this.state

    return (
      <div className={cn('App', { 'App__data-loaded': isDataLoaded })}>
        <Header />
        <main className="App__container">
          <Drawer opened={isDataLoaded} onClear={this.onDataClear} />
          {isDataLoaded ? <div>Data is loaded</div> : <UploadForm onLoad={this.onDataLoad} />}
        </main>
      </div>
    )
  }
}

export default App
