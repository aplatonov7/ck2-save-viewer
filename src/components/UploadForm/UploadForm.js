import React, { Component } from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import LoadingBar from '../LoadingBar'
import dataManager from '../../dataManager'

import './UploadForm.css'

class UploadForm extends Component {
  constructor(...props) {
    super(...props)

    this.state = {
      loading: false,
      progress: null,
      filepath: null,
    }
  }

  onProgress = progress => this.setState({ progress })

  readFile = e => {
    const file = e.target.files[0]
    if (!file) return

    const filepath = e.target.value.split('\\').slice(-1)
    this.setState(() => ({ loading: true, filepath }))

    const reader = new FileReader()
    reader.onload = this.handleLoad
    reader.readAsText(file)
  }

  handleLoad = e => {
    const content = e.target.result
    dataManager.loadData(content, this.onProgress).then(() => {
      this.setState(() => ({ loading: false, progress: null, filepath: null }))
      if (this.props.onLoad) this.props.onLoad()
    })
  }

  render() {
    const { loading, progress, filepath } = this.state
    return (
      <div className={cn('UploadForm__container', { 'UploadForm__container-loading': loading })}>
        {this.state.loading ? (
          [
            <div key="1" className="UploadForm__caption">
              Uploading and parsing {filepath}
            </div>,
            <small key="2" className="UploadForm__sub-caption">
              This might take quite a while (-_-) zzz
            </small>,
            <LoadingBar key="3" value={progress} />,
          ]
        ) : (
          <label className="UploadForm__label" htmlFor="saveInput">
            Click to upload your save file
            <input
              className="UploadForm__input"
              id="saveInput"
              type="file"
              onChange={this.readFile}
            />
          </label>
        )}
      </div>
    )
  }
}

UploadForm.propTypes = {
  onLoad: PropTypes.func.isRequired,
}

export default UploadForm
