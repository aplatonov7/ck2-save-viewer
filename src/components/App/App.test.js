import React from 'react'
import App from './App'

import Drawer from '../Drawer'
import Header from '../Header'
import UploadForm from '../UploadForm'

describe('App component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should not have data loaded by default', () => {
    expect(wrapper.state().isDataLoaded).toBe(false)
  })

  it('should render Header', () => {
    expect(wrapper.find(Header).length).toEqual(1)
  })

  it('should render UploadForm with correct props', () => {
    const uploadForm = wrapper.find(UploadForm)
    expect(uploadForm.length).toEqual(1)
    expect(typeof uploadForm.props().onLoad).toBe('function')
    uploadForm.props().onLoad()
    expect(wrapper.state().isDataLoaded).toBe(true)
  })

  it('should not render UploadForm if data is loaded', () => {
    wrapper.setState({ isDataLoaded: true })
    expect(wrapper.find(UploadForm).length).toEqual(0)
  })

  it('should render Drawer with correct props', () => {
    let drawer = wrapper.find(Drawer)
    expect(drawer.length).toEqual(1)
    expect(drawer.props().opened).toBe(false)
    expect(typeof drawer.props().onClear).toBe('function')
    wrapper.setState({ isDataLoaded: true })
    drawer = wrapper.find(Drawer)
    expect(drawer.props().opened).toBe(true)
    drawer.props().onClear()
    expect(wrapper.state().isDataLoaded).toBe(false)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
