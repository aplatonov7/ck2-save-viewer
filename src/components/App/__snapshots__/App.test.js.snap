// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="Header"
  >
    CK2 save to stats
  </header>
  <main
    className="App__container"
  >
    <aside
      className="Drawer"
    >
      <div
        className="Drawer__row"
      >
        Data is loaded
        <button
          className="Drawer__delete-btn"
          onClick={[Function]}
          title="Delete current data so you can upload another file"
        >
          Delete data
        </button>
      </div>
    </aside>
    <div
      className="UploadForm__container"
    >
      <label
        className="UploadForm__label"
        htmlFor="saveInput"
      >
        Click to upload your save file
        <input
          className="UploadForm__input"
          id="saveInput"
          onChange={[Function]}
          type="file"
        />
      </label>
    </div>
  </main>
</div>
`;
